### DISCUSSION TITLE
windows driver's types support

### DISCUSSION THREAD

**Discussion by nitrogar** [Vector35/binaryninja-api#2583] [Q&A] [answered]
Hi, dose Binary Ninja support windows driver types like DRIVER_OBJECT ??
if no how can I import it ??

**Comment by psifertex** [comment, marked as answer]
Very sorry for the delay on replying here, a few discussions have slipped through the cracks. 

Right now, no, this is a part of the current 3.1 windows release plan though to both include the built-in windows types and additionally, we're [adding clang type parsing](https://github.com/Vector35/binaryninja-api/issues/3007) which will mean that you'll be able to directly just import `windows.h`. Unfortunately, neither of those exist quite yet. 

To be clear, you could manually define the types now using the existing type UI and add new types but you have to deal with the large amounts of defines and dependencies that exist in the windows type ecosystem which is what we plan to make not an issue with the [3.1 release](https://github.com/Vector35/binaryninja-api/milestone/11).
